cmake/SetupRajaConfig.cmake

```diff
@@ -87,3 +87,39 @@ if (RAJA_TIMER STREQUAL "cycle")
 else ()
     set(RAJA_USE_CYCLE   OFF CACHE BOOL "Use clock_gettime for timer"       )
 endif ()
+
+# Configure a header file with all the variables we found.
+configure_file(${PROJECT_SOURCE_DIR}/include/RAJA/config.hxx.in
+  ${PROJECT_BINARY_DIR}/include/RAJA/config.hxx)
+
+# Configure CMake config
+configure_file(${PROJECT_SOURCE_DIR}/share/raja/cmake/RAJA-config.cmake.in
+  ${PROJECT_BINARY_DIR}/share/raja/cmake/raja-config.cmake)
+
+install(FILES ${PROJECT_BINARY_DIR}/share/raja/cmake/raja-config.cmake
+  DESTINATION share/raja/cmake/)
+
+# Setup pkg-config
+include(FindPkgConfig QUIET)
```

Comment:
Does this always do the presence check? I'm used to seeing it as `find_package(PkgConfig)`.  Aside from that, this all looks like an improvement.